source/common/network/address_impl.cc

```diff
@@ -61,16 +73,8 @@ StatusOr<Address::InstanceConstSharedPtr> addressFromSockAddr(const sockaddr_sto
     const struct sockaddr_in6* sin6 = reinterpret_cast<const struct sockaddr_in6*>(&ss);
     ASSERT(AF_INET6 == sin6->sin6_family);
     if (!v6only && IN6_IS_ADDR_V4MAPPED(&sin6->sin6_addr)) {
```

Comment:
I think you can just call v4CompatibleAddress() here and if null do the other branch?

Reply:
Emm...the Instance obj doesn't create yet. We create different Instance obj below these few lines and depend on this check result.

https://github.com/envoyproxy/envoy/pull/19716/files/228cc7e14615516b1397e3501f8dfa66cc5676a1#diff-6925fefb8a60a68e4059b0d4fdbb943c4b3af9fa77df231724cabcc1f1b35f9fL74-L76